syntax = "proto3";
package cas.proto;

import "Mesh.proto";
import "SoundSource.proto";
import "Temperature.proto";
import "BotCar.proto";
import "BotGripper.proto";
import "BotMotor.proto";

message DataMessage {
    enum Type {
        MESH = 0;
        SOUND_SOURCE = 1;
        TEMPERATURE = 2;
        BOT_CAR = 3;
        BOT_GRIPPER = 4;
        BOT_MOTOR = 5;
    }
    Type type = 1;
    oneof data {
        Mesh mesh = 2;
        SoundSource sound_source = 3;
        Temperature temperature = 4;
        BotCar bot_car = 5;
        BotGripper bot_gripper = 6;
        BotMotor bot_motor = 7;
    }
}